src/hotspot/share/prims/forte.cpp

```diff
@@ -323,7 +323,7 @@ static bool find_initial_Java_frame(JavaThread* thread,
     // See if we can find a useful frame
     int loop_count;
     int loop_max = MaxJavaStackTraceDepth * 2;
-    RegisterMap map(thread, false);
+    RegisterMap map(thread, false, false);
```

Comment:
Can we add some comments as to what the false parameters mean please.

RegisterMap map(thread, false /* no update */, false /*no stackwatermark frame processing */);

Though it may be that a more elaborate block comment is needed to explain why we don't want stackwatermark frame processing.

Reply:
Let me check with Erik if it makes sense to put more generic comments about the usage of stackwatermark frame processing in RegisterMap, frames etc.  They can't be updated in an arbitrary thread state. It makes sense describe this info in stackwatermarking.

Reply:
> Let me check with Erik if it makes sense to put more generic comments about the usage of stackwatermark frame processing in RegisterMap, frames etc.  They can't be updated in an arbitrary thread state. It makes sense describe this info in stackwatermarking.

You could say something generic like "StackWatermark can only be used when at points where the stack can be parsed by the GC", or something like that.

Reply:
Thank you for your prompt response. I meant that it makes sense to update comments for RegisterMap to mention this. Currently, the doc says how to use it and how to disable update:
"Updating of the RegisterMap can be turned off by instantiating the
//      register map as: RegisterMap map(thread, false);"
But it says nothing about why and how  process_frames should be set.
It might make sense to put this info there so anyone could easily find and read it. I think it is better to put it there rather than in forte.cpp.

Reply:
> Thank you for your prompt response. I meant that it makes sense to update comments for RegisterMap to mention this. Currently, the doc says how to use it and how to disable update:
> 
> "Updating of the RegisterMap can be turned off by instantiating the
> 
> //      register map as: RegisterMap map(thread, false);"
> 
> But it says nothing about why and how  process_frames should be set.
> 
> It might make sense to put this info there so anyone could easily find and read it. I think it is better to put it there rather than in forte.cpp.

Yes I agree - that does make sense.